Python 3 support

kubo-von opened the pull request:
Hey @andrewkaufman,
I've added the PYTHON_VERSION arg. 
Tested to build fine with gaffer-0.59.4.0-linux-python3, cannot verify with python2 because it fails for me with `/usr/bin/ld: cannot find -lboost_python27` but it does fail the same way with your master too though, so I assume it's a problem somewhere on my side..
Cheers

kubo-von committed:
cmake edits to adjust for gaffer 0.59, python3

fixing python code to run with Python 3

Adding cmake option to specify python version with PYTHON_VERSION arg

Adding cmake option to specify python version with PYTHON_VERSION arg

Readme Python 3 note

Removing nasty build folder + build.sh

cmake using python version directly

readme update

andrewkaufman commented on CMakeLists.txt:
```diff
@@ -25,11 +25,20 @@ target_compile_definitions( DemoGafferExtensionModule PRIVATE BOOST_SIGNALS_NO_D
 
 if(APPLE)
-  target_include_directories( DemoGafferExtensionModule PRIVATE include ${GAFFER_ROOT}/include ${GAFFER_ROOT}/lib/Python.framework/Versions/2.7/include/python2.7 )
+  if( "${PYTHON_VERSION}" MATCHES "3.7" )
+    target_include_directories( DemoGafferExtensionModule PRIVATE include ${GAFFER_ROOT}/include ${GAFFER_ROOT}/lib/Python.framework/Versions/3.7/include/python3.7 )
+  else()
+    target_include_directories( DemoGafferExtensionModule PRIVATE include ${GAFFER_ROOT}/include ${GAFFER_ROOT}/lib/Python.framework/Versions/2.7/include/python2.7 )
+  endif()
```
This should be possible without an `if` by just using `${PYTHON_VERSION}` in the path directly, right?

andrewkaufman commented on CMakeLists.txt:
```diff
@@ -25,11 +25,20 @@ target_compile_definitions( DemoGafferExtensionModule PRIVATE BOOST_SIGNALS_NO_D
 else()
-  target_include_directories( DemoGafferExtensionModule PRIVATE include ${GAFFER_ROOT}/include ${GAFFER_ROOT}/include/python2.7 )
-  target_link_libraries( DemoGafferExtensionModule GafferBindings IECorePython boost_python27 python2.7)
+  if( "${PYTHON_VERSION}" MATCHES "3.7" )
+    target_include_directories( DemoGafferExtensionModule PRIVATE include ${GAFFER_ROOT}/include ${GAFFER_ROOT}/include/python3.7m )
+    target_link_libraries( DemoGafferExtensionModule GafferBindings IECorePython boost_python37 python3.7m)
+  else()
+    target_include_directories( DemoGafferExtensionModule PRIVATE include ${GAFFER_ROOT}/include ${GAFFER_ROOT}/include/python2.7 )
+    target_link_libraries( DemoGafferExtensionModule GafferBindings IECorePython boost_python27 python2.7)
+  endif()
```
As above, but maybe a bit trickier... I'm no cmake expert, but I'd hope there is a way to strip the `.` from ${PYTHON_VERSION}

andrewkaufman reviewed (commented):
Thanks @kubo-von. I made some non-essential comments, but it'd be nice to address if you know how.

Could you also rebase to avoid the accidental file additions/deletions?

kubo-von commented:
Hey @andrewkaufman 
I've addressed your comments and (hopefully correcly) rebased. There's the small issue with python folder in gaffer having "m" at the end for python 3.7 let me know if you wanna handle that differently.
Cheers

andrewkaufman commented:
@kubo-von thanks for the contribution!

kubo-von commented:
> @kubo-von thanks for the contribution!

My pleasure!
